index.bs

```diff
@@ -717,6 +717,56 @@ is done in a consistent and privacy-friendly way:
      stable id during the session for the given origin, but also be able to deterministically produce
      the same id in subsequent sessions.
 
+<h3 id="wrapper-apis">Wrapper API considerations</h3>
+
+Many modern operating systems come with convenient APIs which abstract away
+normally complex technology in the form of user-friendly APIs. Exposing these
+APIs to the web platform is getting more and more common, but you need to be
+careful when these are being exposed to the web platform.
+
+ : Generalize interface when underlying API is overly specific
+ :: If the wrapped native API is extremely specific to a certain library, it is
+    recommended to design a generic API which allows different browsers to not
+    have to implement this by emulating behavior in the event a different underlying
+    native API or library is used.
+
+    A common example of bad practice is exposing the exact lifecycle and structs
+    of the underlying native API.
+
+ : Make API promise based whenever it makes sense
+ :: Even if the underlying API is synchronous, this does not necessarily mean it
+    translates well when ported to the web platform. Exposing a synchronous native
+    API to the web platform is generally considered bad taste, and should be
```

Comment:
"generally considered bad taste" makes it sound like there aren't good reasons behind it... but I think there are